diffrax/_solver/srk.py

```diff
@@ -0,0 +1,740 @@
+        return self.coeffs_kk.a
+
+    @property
+    def b_kk(self):
+        """The coefficient for the space-time-time Lévy area when computing
+        the output."""
+        return self.coeffs_kk.b
+
+    @property
+    def get_b_error_kk(self) -> Optional[Float[np.ndarray, " s"]]:
+        if isinstance(self.coeffs_kk, GeneralCoeffsWithError):
+            return self.coeffs_kk.b_error
+        else:
+            return None
+
+
+class StochasticTableau(AbstractStochasticTableau[COEFFS]):
+    coeffs_w: COEFFS
+
+    def check(self):
+        return self.coeffs_w.check()
+
+
+class SpaceTimeLATableau(_AbstractSTLATableau[COEFFS]):
+    coeffs_w: COEFFS
+    coeffs_hh: COEFFS
+
+    def check(self):
+        w_num_stages = self.coeffs_w.check()
+        assert w_num_stages == self.coeffs_hh.check()
+        return w_num_stages
+
+
+class SpaceTimeTimeLATableau(_AbstractSTTLATableau[COEFFS]):
+    coeffs_w: COEFFS
+    coeffs_hh: COEFFS
+    coeffs_kk: COEFFS
+
+    def check(self):
+        w_num_stages = self.coeffs_w.check()
+        assert w_num_stages == self.coeffs_hh.check() == self.coeffs_kk.check()
+        return w_num_stages
+
+
+@dataclass(frozen=True)
+class StochasticButcherTableau:
+    """A Butcher Tableau for Stochastic Runge-Kutta methods."""
+
+    # Only supports explicit SRK so far
+    c: np.ndarray
+    b_sol: np.ndarray
+    b_error: Optional[np.ndarray]
+    a: list[np.ndarray]
+
+    # Coefficients for the Brownian increment
+    cfs_bm: AbstractStochasticTableau
+
+    # For some stages we may not need to evaluate the vector field for both
+    # the drift and the diffusion. This avoids unnecessary computations.
+    ignore_stage_f: Optional[np.ndarray] = None
+    ignore_stage_g: Optional[np.ndarray] = None
+
+    @property
+    def additive_noise(self):
+        return self.cfs_bm.additive_noise
+
+    def __post_init__(self):
+        assert self.c.ndim == 1
+        for a_i in self.a:
+            assert a_i.ndim == 1
+        assert self.b_sol.ndim == 1
+        assert (self.b_error is None) or self.b_error.ndim == 1
+        assert self.c.shape[0] == len(self.a)
+        assert all(i + 1 == a_i.shape[0] for i, a_i in enumerate(self.a))
+        assert (self.b_error is None) or self.b_error.shape[0] == self.b_sol.shape[0]
+        assert self.c.shape[0] + 1 == self.b_sol.shape[0]
+
+        assert np.allclose(sum(self.b_sol), 1.0)
+
+        assert self.cfs_bm.check() == self.b_sol.shape[0]
+
+        if self.b_error is not None and (not self.additive_noise):
+            assert self.cfs_bm.has_error_estimate
+
+        if self.ignore_stage_f is not None:
+            assert len(self.ignore_stage_f) == len(self.b_sol)
+        if self.ignore_stage_g is not None:
+            assert len(self.ignore_stage_g) == len(self.b_sol)
+
+
+StochasticButcherTableau.__init__.__doc__ = """**Arguments:**
+
+Let `s` denote the number of stages of the solver.
+
+- `s`: The number of stages of the solver.
```

Comment:
I don't think this is an argument?

The other arguments are out-of-order with their definition above.

Reply:
Oops, I forgot to double check what copilot wrote.

Reply:
xD